src/plugins/data_views/server/register_index_pattern_usage_collection.ts

```diff
@@ -80,60 +85,74 @@ export async function getIndexPatternTelemetry(indexPatterns: DataViewsContract)
     },
   };
 
-  await ids.reduce(async (col, id) => {
-    await col;
-    const ip = await indexPatterns.get(id);
+  const findOptions: SavedObjectsCreatePointInTimeFinderOptions = {
+    type: DATA_VIEW_SAVED_OBJECT_TYPE,
+    perPage: 1000,
+    fields: ['attributes.fields', 'attributes.runtimeFieldMap'],
+  };
 
-    const scriptedFields = ip.getScriptedFields();
-    const runtimeFields = ip.fields.filter((fld) => !!fld.runtimeField);
+  const finder = savedObjectsService.createPointInTimeFinder<DataViewFieldAttrs>(findOptions);
 
-    if (scriptedFields.length > 0) {
-      // increment counts
-      results.indexPatternsWithScriptedFieldCount++;
-      results.scriptedFieldCount += scriptedFields.length;
+  for await (const response of finder.find()) {
+    const { saved_objects: savedObjects, total } = response;
+    results.indexPatternsCount = total;
 
-      // calc LoC
-      results.perIndexPattern.scriptedFieldLineCount = minMaxAvgLoC(
-        scriptedFields.map((fld) => fld.script || '')
-      );
-
-      // calc field counts
-      results.perIndexPattern.scriptedFieldCount.min = updateMin(
-        results.perIndexPattern.scriptedFieldCount.min,
-        scriptedFields.length
-      );
-      results.perIndexPattern.scriptedFieldCount.max = updateMax(
-        results.perIndexPattern.scriptedFieldCount.max,
-        scriptedFields.length
-      );
-      results.perIndexPattern.scriptedFieldCount.avg =
-        results.scriptedFieldCount / results.indexPatternsWithScriptedFieldCount;
-    }
-
-    if (runtimeFields.length > 0) {
-      // increment counts
-      results.indexPatternsWithRuntimeFieldCount++;
-      results.runtimeFieldCount += runtimeFields.length;
+    savedObjects.forEach((obj) => {
+      const fields = JSON.parse(obj.attributes.fields) || [];
```

Comment:
the JSON.parse threw an exception locally, so it's recommendable to use a`try{}catch(){}` pattern for this case

Reply:
Ah, I had failed to push a change. I also changed the default data view definition in the test not to have the attributes that are used as to provide test coverage.